Installation issue: amdlibflame

`amdlibflame` doesn't have a build dependency on `python +pythoncmd` so it fails to build when a `python` binary/symlink isn't present.

### Steps to reproduce the issue

```console
$ spack install amdlibflame
(...)
==> Installing amdlibflame-3.0-bycjukuru44crh5dkhekz2ftp6p3yefy
==> No binary for amdlibflame-3.0-bycjukuru44crh5dkhekz2ftp6p3yefy found: installing from source
==> Using cached archive: /home/bc/spack/var/spack/cache/_source-cache/archive/d9/d94e08b688539748571e6d4c1ec1ce42732eac18bd75de989234983c33f01ced.tar.gz
==> Applied patch /home/bc/spack/var/spack/repos/builtin/packages/amdlibflame/cray-compiler-wrapper.patch
==> amdlibflame: Executing phase: 'autoreconf'
==> amdlibflame: Executing phase: 'configure'
==> amdlibflame: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'V=1'

1 error found in build log:
     248    You may now run 'make' to build all libflame libraries and then 'make install'
     249    to install the libraries.
     250    
     251    ==> amdlibflame: Executing phase: 'build'
     252    ==> [2021-09-09-20:32:26.584483] 'make' '-j16' 'V=1'
     253    (huge line here...)
  >> 254    /usr/bin/env: 'python': No such file or directory
     255    make: *** [Makefile:462: include/x86_64-unknown-linux-gnu/FLAME.h] Error 127
```

### Information on your system

```console
$ spack debug report
* **Spack:** 0.16.2-4209-9084ad69b4
* **Python:** 3.6.8
* **Platform:** linux-rocky8-zen2
* **Concretizer:** original
```

Rocky 8.4 python externals in packages.yaml:
```
  python:
    externals:
    - spec: python@2.7.18+bz2+ctypes+dbm~lzma+nis+pyexpat+readline+sqlite3+ssl~tix~tkinter+uuid+zlib
      prefix: /usr
    - spec: python@3.6.8+bz2+ctypes+dbm+lzma+nis+pyexpat+readline+sqlite3+ssl~tix~tkinter+uuid+zlib
      prefix: /usr
```

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/7140267/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/7140268/spack-build-env.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/amdlibflame/package.py
+++ b/var/spack/repos/builtin/packages/amdlibflame/package.py
@@ -44,7 +44,7 @@ class Amdlibflame(LibflameBase):
 
     provides('flame@5.2', when='@2:')
 
-    depends_on('python', type='build')
+    depends_on('python+pythoncmd', type='build')
 
     @property
     def lapack_libs(self):

--- a/var/spack/repos/builtin/packages/amdlibflame/package.py
+++ b/var/spack/repos/builtin/packages/amdlibflame/package.py
@@ -44,6 +44,8 @@ class Amdlibflame(LibflameBase):
 
     provides('flame@5.2', when='@2:')
 
+    depends_on('python', type='build')
+
     @property
     def lapack_libs(self):
         """find lapack_libs function"""

